buildSrc/src/main/groovy/com/uber/okbuck/core/model/android/AndroidLibTarget.groovy

```diff
@@ -22,4 +26,23 @@ class AndroidLibTarget extends AndroidTarget {
     ManifestMerger2.MergeType getMergeType() {
         return ManifestMerger2.MergeType.LIBRARY
     }
+
+    List<String> getKotlincArguments() {
+        ImmutableList.Builder<String> extraKotlincArgs = ImmutableList.<String>builder()
+        if (hasKotlinAndroidExtensions) {
+            String options = ""
+            getResDirs().each {  String dir ->
+                // :root:module -> /root/module/
+                String module = project.path.replace(":", "/") + "/"
+                // src/main/res -> main
+                String variant = dir.substring(dir.indexOf("/") + 1, dir.lastIndexOf("/"))
```

Comment:
Variant logic is not sound. Any arbitrary directory can be added as a resource directory to a variant.

The variant name should be already available by directly using the `name` field of this target

Reply:
I think the issue here is that the `name` field will always return a singular `debug` or `main` for each resource directory. I have a case where I need each variant name returned with their appropriate resource path. This way a file in `debug` can reference a resource in `main` and kotlin android extensions appropriately generates for that case.

Reply:
For example: My test code uses `import kotlinx.android.synthetic.main.test_layout.view.*` because the path to the resource is `src/main/res`. `name` in that case is always returning `debug`.

Reply:
Getting the available resource dirs that are part of the input sourceset is the right way to go then

But I am curious if the variant name really matters? Is there some sort of precedence/ordering being taken into consideration by the kotlin extension here?

How does it work if I have a folder like `custom_resources` added as a resource folder?

Reply:
I added another example that hopefully makes it clear